src/foundry/client/data/documents/setting.d.mts

```diff
@@ -375,7 +376,7 @@ declare global {
     /**
      * @privateRemarks This exists to let ts know that this class has a private property
      */
-    static #PRIMITIVE_TYPES: any;
+    static #PRIMITIVE_TYPES: [typeof String, typeof Number, typeof Boolean, typeof Array, typeof Symbol, typeof BigInt];
```

Comment:
We should probably be consistent with private properties. Specifically `#theCurrentClass'sName: true` is all that's necessary unless there's some variance implication.